packages/server/test/course.integration.ts

```diff
@@ -355,4 +361,78 @@ describe('Course Integration', () => {
       expect(ucm).toBeUndefined();
     });
   });
+
+  describe('GET /courses/:id/ta_check_in_times', () => {
+    it('tests that events within date range are gotten', async () => {
+      const now = new Date();
+      const yesterday = new Date();
+      yesterday.setUTCHours(now.getUTCHours() - 24);
+
+      const course = await CourseFactory.create();
+      const ta = await UserFactory.create();
+      const professor = await UserFactory.create();
+      await UserCourseFactory.create({
+        user: ta,
+        role: Role.TA,
+        course,
+      });
+      await UserCourseFactory.create({
+        user: professor,
+        role: Role.PROFESSOR,
+        course,
+      });
+
+      await EventFactory.create({
+        user: ta,
+        course: course,
+        time: yesterday,
+        eventType: EventType.TA_CHECKED_IN,
+      });
+
+      await EventFactory.create({
+        user: ta,
+        course: course,
+        time: new Date(Date.now() - 80000000),
+        eventType: EventType.TA_CHECKED_OUT,
+      });
+
+      await EventFactory.create({
+        user: ta,
+        course: course,
+        time: new Date(Date.now() - 70000000),
+        eventType: EventType.TA_CHECKED_IN,
+      });
+
+      const todayAtMidnight = new Date();
+      todayAtMidnight.setHours(0, 0, 0, 0);
+
+      await EventFactory.create({
+        user: ta,
+        course: course,
+        time: todayAtMidnight,
+        eventType: EventType.TA_CHECKED_OUT_FORCED,
+      });
+
+      await EventFactory.create({
+        user: ta,
+        course: course,
+        time: new Date(Date.now() - 1000),
+        eventType: EventType.TA_CHECKED_IN,
+      });
+
+      const data = await supertest({ userId: professor.id })
+        .get(`/courses/${course.id}/ta_check_in_times`)
+        .query({
+          startDate: new Date(Date.now() - 90000000),
+          endDate: new Date(),
+        })
+        .expect(200);
+
+      const checkinTimes = ((data.body as unknown) as TACheckinTimesResponse)
```

Comment:
just curious, what's up with this double type cast here?

Reply:
I wish I knew as well? It got mad at me without first going to `unknown`